### PR TITLE
Static code analysis updates

### PR DESCRIPTION
**Pull request by danchen**
- Fixed javadoc warnings
- Fixed minor typos and variable names
- Updates from static code analysis (mostly on test classes)
  - Remove redundant toString and boxing
  - Finals
  - Localized fields where only used in a single method
  - Replace with Collections.singletonList for single-item ArrayLists

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
+target/
+test-output/
+.settings
 .classpath
 .checkstyle
-/test-output
 .idea/
-*.iml
+.sonar/
+*.iml
```
Comment (DeathByTape) [inline .gitignore]: new line

File: elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java [hunk 2]
```diff
@@ -135,7 +135,7 @@ public String getBinding(Class<?> entityClass) {
     /**
      * Get the list of relationship names for an entity
```
Comment (DeathByTape) [inline elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java]: Just noticing with the other changes: we don't need `.` here?

**Comment by DeathByTape** [comment]
Actual code changes look good to me. Just two styling comments. :+1:

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by danchen** [comment]
@DeathByTape resolved your comments.

**Comment by DeathByTape** [comment]
:+1: looks great!

**Pull request merged**